### PR TITLE
(DOCSP-12468): Update OAuth provider login docs

### PR DESCRIPTION
**Pull request by nlarew**
### Issue JIRA link:

(DOCSP-12468): Write explicit Documentation around how to Acquire Tokens for each OAuth provider

### Docs staging link (requires sign-in on MongoDB Corp SSO):

- [Authenticate a User (Android) > Facebook OAuth](https://docs.mongodb.com/realm/android/authenticate/#facebook-oauth)
- [Authenticate a User (Android) > Google OAuth](https://docs.mongodb.com/realm/android/authenticate/#google-oauth)
- [Authenticate a User (iOS) > Facebook OAuth](https://docs.mongodb.com/realm/ios/authenticate/#facebook-oauth)
- [Authenticate a User (iOS) > Google OAuth](https://docs.mongodb.com/realm/ios/authenticate/#google-oauth)

### CODE REVIEW

File: examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java [hunk 1]
```diff
@@ -178,30 +178,40 @@ public void testCustomJWT() {
...
-            app.loginAsync(facebookCredentials, it -> {
-                // :hide-start:
-                Assert.assertEquals(false, it.isSuccess());
-                // :hide-end:
-                if (it.isSuccess()) {
-                    Log.v("AUTH", "Successfully authenticated using Facebook OAuth.");
-                    user.set(app.currentUser());
-                } else {
-                    Log.e("AUTH", it.getError().toString());
+            .build());
```
Comment (ccho-mongodb) [inline examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java:183]: Strange indentation, did you mean to change it?

File: examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java [hunk 2]
```diff
@@ -178,30 +178,40 @@ public void testCustomJWT() {
...
+                    @Override
+                    public void onSuccess(LoginResult loginResult) {
+                        // Signed in successfully, forward credentials to MongoDB Realm.
+                        val accessToken = loginResult.getAccessToken();
+                        val facebookCredentials: Credentials = Credentials.facebook(accessToken);
+                        app.loginAsync(facebookCredentials, it -> {
+                            if (it.isSuccess) {
+                                Log.v(TAG, "Successfully logged in to MongoDB Realm using Facebook OAuth.")
+                            } else {
+                                Log.e(TAG, "Error logging in to MongoDB Realm: ${it.error.toString()}")
```
Comment (ccho-mongodb) [inline examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java:196]: Question: is it possible to bubble up an exception or return code to signify that it failed the loginAsync? Or is the log message sufficient for real-world use?

**Review by ccho-mongodb — APPROVED** [review]
LGTM! Did not check the code examples carefully, but the first two examples looked good.

File: examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java (hunk 1 above)
Comment (nlarew) [reply examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java:183]: Nope good catch!

File: examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java (hunk 2 above)
Comment (nlarew) [reply examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java:196]: I'm not sure why it wouldn't be possible. Historically we haven't heard complaints about this style (i.e. indicating the branch condition with logs) but I can try to update the log message to be a bit clearer that it failed in loginAsync vs the FB portion of login

Comment (nlarew) [reply examples/android/app/src/androidTest/java/com/mongodb/realm/examples/java/AuthenticationTest.java:196]: I think we can log the error too which should provide a stack trace for additional context. Good call!

**Pull request merged**